pkg/build/api/types.go

```diff
@@ -278,9 +259,6 @@ type SourceControlUser struct {
 
 // BuildStrategy contains the details of how to perform a build.
 type BuildStrategy struct {
-	// Type is the kind of build strategy.
-	Type BuildStrategyType
```

Comment:
@bparees just to make it clear, are we removing this in favor of checking which one and only one of the three other fields is non-nil?